src/classes/HiFiMixerSession.ts

```diff
@@ -536,6 +503,38 @@ export class HiFiMixerSession {
                 this.onUserDataUpdated(allNewUserData);
             }
         }
+
+        if (jsonData.deleted_visit_ids) {
+            let allDeletedUserData: Array<ReceivedHiFiAudioAPIData> = [];
+
+            let deletedVisitIDs = jsonData.deleted_visit_ids;
+            for (const deletedVisitID of deletedVisitIDs) {
+                let hashedVisitID = deletedVisitID;
+
+                let deletedUserData = new ReceivedHiFiAudioAPIData({
+                    hashedVisitID: hashedVisitID
+                });
+
+                let mixerPeerKeys = Object.keys(this._mixerPeerKeyToStateCacheDict);
+                for (const key of mixerPeerKeys) {
+                    let peerData = this._mixerPeerKeyToStateCacheDict[key];
+                    if (peerData.hashedVisitID === hashedVisitID) {
+                        if (peerData.providedUserID) {
+                            deletedUserData.providedUserID = peerData.providedUserID;
+                        }
+                        delete this._mixerPeerKeyToStateCacheDict[key];
```

Comment:
Note: I'm correctly deleting the peer from the map on delete.  Previously this operation was commented out with a TODO comment about it (see the above removed block).

Reply:
thank you!